### ISSUE TITLE
Updating an object leads to duplicate quads

### ISSUE THREAD

**Issue by denisvmedia** [cayleygraph/cayley#756]
Here is an example of my code:
https://gist.github.com/denisvmedia/6897cf959bdb7aa5830f68884b03730a

**Description**

In my code I need to keep some object and then to be able to update one or more properties of that. It works, but it has an issue: the updated field leads to a duplicated quad.

**Steps to reproduce the issue:**
1. compile and run [this example.](https://gist.github.com/denisvmedia/6897cf959bdb7aa5830f68884b03730a)

**Received results:**

```
quads:
<gc:cluster1> -- <rdf:type> -> <gc:Cluster>
<gc:cluster1> -- <gc:name> -> "Cluster1"
<gc:host1> -- <rdf:type> -> <gc:Host>
<gc:host1> -- <gc:name> -> "Host1"
<gc:host1> -- <gc:cluster> -> <gc:cluster1>
<gc:host1> -- <deploymentStatus> -> "ok"
<gc:host1> -- <deploymentStatus> -> "error"
```

(note that `<gc:host1> -- <deploymentStatus>` is repeated two times with the original and the update values).

**Expected results:**

```
quads:
<gc:cluster1> -- <rdf:type> -> <gc:Cluster>
<gc:cluster1> -- <gc:name> -> "Cluster1"
<gc:host1> -- <rdf:type> -> <gc:Host>
<gc:host1> -- <gc:name> -> "Host1"
<gc:host1> -- <gc:cluster> -> <gc:cluster1>
<gc:host1> -- <deploymentStatus> -> "error"
```

**Cayley and DB backend versions:**

```
github.com/boltdb/bolt v1.3.1
github.com/cayleygraph/cayley v0.7.5
```

P.S. Well, I know, I can delete the quad manually, but is that a proper way?

**Comment by dennwc** [comment]
Yes, it won't remove old data automatically. You'd have to remove an old value manually.

As an alternative, it may be possible to implement a "diff writer" that will accept and old state and a new one and will execute add/remove commands accordingly. Although this will be harder.
